Userland/Libraries/LibGfx/ICC/Profile.cpp

```diff
@@ -865,17 +865,34 @@ ErrorOr<void> Profile::check_tag_types()
     // Profile ID of /System/Library/ColorSync/Profiles/ITU-2020.icc on macOS 13.1.
     static constexpr Crypto::Hash::MD5::DigestType apple_itu_2020_id = { 0x57, 0x0b, 0x1b, 0x76, 0xc6, 0xa0, 0x50, 0xaa, 0x9f, 0x6c, 0x53, 0x8d, 0xbe, 0x2d, 0x3e, 0xf0 };
 
+    auto has_type = [&](auto tag, std::initializer_list<TagTypeSignature> types, std::initializer_list<TagTypeSignature> v4_types) {
+        if (auto type = m_tag_table.get(tag); type.has_value()) {
+            auto type_matches = [&](auto wanted_type) { return type.value()->type() == wanted_type; };
+            return any_of(types, type_matches) || (is_v4() && any_of(v4_types, type_matches));
+        }
+        return true;
```

Comment:
Just skimming over this; why does `has_type` return `true` if the type is not in the table at all?

Reply:
Because that way I can write

```
if (!has_type(some_tag, {some_type}, {}))
    return Error;
```

and it does what I want.

The semantics are supposed to be: If I have a tag with that name, then it should have the given type. If I don't have that tag at all, I can't check it's type, and it's all good.

(A different function further up checks if required tags are present or not.)

I also thought this looked a bit weird! But I couldn't come up with anything better, and I didn't find it _that_ weird. Do you have any suggestions on de-weirding it? :)

Reply:
I guessed that that was the case! Maybe explain it more in the name of the function, `optional_type_is_any_of` for example?

Reply:
Yeah it's a very specific local functionality that's definitely not worth discussing naming about in such great detail ;-)